### DISCUSSION TITLE
Reset Environment with service Principle

### DISCUSSION THREAD

**Discussion by h-karthik12** [microsoft/powerplatform-build-tools#526] [Q&A]
I tried to reset the env with service principle and i have added the SP as the system Admin. When i run the
 pac admin reset --environment https://envurl.com
i get this error
Error: The service principal with id '8ce***' for application 3afcf***  does not have permission to access the path 'https://10.0.7.17:20054/providers/Microsoft.BusinessAppPlatform/scopes/admin/environments?`$expand=permissions&api-version=2020-08-01' in tenant 764****.

Not sure what is that i am missing. I can do the same with normal user. Is this feature not available for SP or some privileges needed to be added?

**Comment by h-karthik12** [comment]
I even tried from Azure DevOps it fails in the Command as well as in the ci-cd tasks

**Comment by tehcrashxor** [comment]
In order for Service Principals to be permitted to talk to the Admin endpoint, they first need to be registered with the Admin endpoint per the ["Registering an admin management application" section of these docs](https://learn.microsoft.com/en-us/power-platform/admin/powerplatform-api-create-service-principal#registering-an-admin-management-application).

Service Principals created by PAC via `pac admin create-service-principal` are registered against that endpoint by default (unless you happened to use PAC v1.26.x due to a bug that has been fixed in v1.27.x)

**Reply by AshV** [reply]
This is in preview. What to use fro production?
